src/components/products/product-subscribe/ko/runtime/product-subscribe.ts

```diff
@@ -152,11 +154,14 @@ export class ProductSubscribe {
 
         try {
             if (this.delegationEnabled) {
-                const delegation = new URLSearchParams();
-                delegation.append(DelegationParameters.ProductId, Utils.getResourceName("products", productId));
-                delegation.append(DelegationParameters.UserId, Utils.getResourceName("users", userId));
-                await this.router.navigateTo(`/${DelegationActionPath.subscribe}?${delegation.toString()}`);
-                return;
+                const delegationParam = {};
+                delegationParam[DelegationParameters.ProductId] =  Utils.getResourceName("products", productId);
```

Comment:
In `delegationParam[DelegationParameters.ProductId] = Utils.getResourceName("products", productId);` can we use `productName` directly, without wrapping it into `productId` and extracting it back with `Utils.getResourceName`?